buildSrc/src/main/kotlin/software/aws/toolkits/gradle/jacoco/RemoteCoverage.kt

```diff
@@ -0,0 +1,104 @@
+// Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.gradle.jacoco
+
+import org.gradle.api.file.RegularFileProperty
+import org.gradle.api.services.BuildService
+import org.gradle.api.services.BuildServiceParameters
+import org.gradle.api.tasks.testing.Test
+import org.gradle.testing.jacoco.plugins.JacocoTaskExtension
+import org.jacoco.core.data.ExecutionData
+import org.jacoco.core.data.ExecutionDataWriter
+import org.jacoco.core.data.IExecutionDataVisitor
+import org.jacoco.core.data.ISessionInfoVisitor
+import org.jacoco.core.data.SessionInfo
+import org.jacoco.core.runtime.RemoteControlReader
+import org.jacoco.core.runtime.RemoteControlWriter
+import java.net.ServerSocket
+import java.net.Socket
+import java.util.concurrent.atomic.AtomicBoolean
+
+class RemoteCoverage private constructor(task: Test) {
+    companion object {
+        fun enableRemoteCoverage(task: Test) = RemoteCoverage(task)
+    }
+
+    private val server = "localhost"
+    private val port = 6300
+
+    init {
+        task.extensions.findByType(JacocoTaskExtension::class.java)?.let {
+            val execFile = it.destinationFile ?: return@let
+            val jacocoServer = task.project.gradle.sharedServices.registerIfAbsent("jacocoServer", JacocoServer::class.java) {
+                parameters.execFile.set(execFile)
+            }
+
+            task.doFirst {
+                jacocoServer.get().start()
+            }
+        } ?: task.logger.warn("$task does not have Jacoco enabled on it")
+    }
+
+    abstract class JacocoServer : BuildService<JacocoServer.Params>, AutoCloseable {
+        interface Params : BuildServiceParameters {
+            val execFile: RegularFileProperty
+        }
+
+        private val serverSocket = ServerSocket(6300) // Default jacoco port
```

Comment:
port is duplicated, probably makes sense as `DEFAULT_JACOCO_PORT`

Reply:
Fixed